File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/sample/search/SampleTypeSearchCriteria.java
/*
 * Copyright 2016 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.search.AbstractEntityTypeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.semanticannotation.search.SemanticAnnotationSearchCriteria;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.sample.search.SampleTypeSearchCriteria")
public class SampleTypeSearchCriteria extends AbstractEntityTypeSearchCriteria
{
    private static final long serialVersionUID = 1L;

    public ListableSampleTypeSearchCriteria withListable()
    {
        return with(new ListableSampleTypeSearchCriteria());
    }

    public SemanticAnnotationSearchCriteria withSemanticAnnotations()
    {
        return with(new SemanticAnnotationSearchCriteria());
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("SAMPLE_TYPE");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/sample/search/SampleTypeSearchCriteria.js
define([ "require", "stjs", "as/dto/entitytype/search/AbstractEntityTypeSearchCriteria", "as/dto/sample/search/ListableSampleTypeSearchCriteria",
		"as/dto/semanticannotation/search/SemanticAnnotationSearchCriteria" ], function(require, stjs, AbstractEntityTypeSearchCriteria) {
	var SampleTypeSearchCriteria = function() {
		AbstractEntityTypeSearchCriteria.call(this);
	};
	stjs.extend(SampleTypeSearchCriteria, AbstractEntityTypeSearchCriteria, [ AbstractEntityTypeSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.sample.search.SampleTypeSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withListable = function() {
			var ListableSampleTypeSearchCriteria = require("as/dto/sample/search/ListableSampleTypeSearchCriteria");
			return this.addCriteria(new ListableSampleTypeSearchCriteria());
		};
		prototype.withSemanticAnnotations = function() {
			var SemanticAnnotationSearchCriteria = require("as/dto/semanticannotation/search/SemanticAnnotationSearchCriteria");
			return this.addCriteria(new SemanticAnnotationSearchCriteria());
		};
	}, {});

	return SampleTypeSearchCriteria;
})